class/defaults.yml

```diff
@@ -47,15 +48,23 @@ parameters:
       rules: []
     # Use Bitnami Postgres installed by the Keycloak chart by default
     database:
-      builtin: true
```

Comment:
It's more about consistency. With the `provider` switch, we can configure parameters per provider. e.g.
```yaml
database:
  provider: external
  external:
    host: ...
```
as of now, the built-in provider only has the `enabled` flag, since all of the built-in values go into `helm_values.postgresql`:
```yaml
database:
  provider: builtin
  builtin:
    enabled: true
```
This syntax is similar to how Kubernetes uses it, e.g. update strategy:
```yaml
Deployment:
...
strategy: 
  type: RollingUpdate
  rollingUpdate:
    ...
```
So I thought it should be fairly familiar and extensible.
Or did you mean something else?

Reply:
The ugly thing *in both cases* is that we still have to disable the built-in provider when using the external provider. This is due to the fact that the `enabled` flag gets used as a Reclass variable in `helm_values.postgresql.enabled` and I'm not sure if we somehow can if-else this in jsonnet or so. It would be nice if the postgres helm chart gets disabled when setting `provider=external`.

@simu thoughts? is that correct?

Reply:
> I'm not sure if we somehow can if-else this in jsonnet or Reclass. It would be nice if the postgres helm chart gets disabled when setting `provider=external`.

The only way to do if-else (really switch-case) in Reclass that I found is with a lookup table:

```
parameters:
  keycloak:
    # make lookup table constant to prevent changes to it through the hierarchy
    =_enable_pg_chart:
      builtin: true
      external: false
    database:
      provider: builtin
    helm_values:
      postgresql:
        enabled: ${keycloak_enable_pg_chart:${keycloak:database:provider}}
```

Reply:
oh that's neat, it works and makes documentation easier!